### PR TITLE
Add stream ext

### PR DESCRIPTION
**Pull request by mathstuf**

### CODE REVIEW

**Comment by mathstuf** [comment]
Cc: @carllerche Followup to #558.

File: tests/timer.rs [hunk 1]
```diff
@@ -71,6 +71,7 @@ fn starving() {
     rx.recv().unwrap();
 }
 
+#[allow(deprecated)]
```
Comment (carllerche) [inline tests/timer.rs]: Could you remove this? I assume you get a warning. We should fix this in a follow up PR by updating the test.

Comment (mathstuf) [reply tests/timer.rs]: I removed the commit which had this.

**Comment by mathstuf** [comment]
Ping?

**Comment by tobz** [comment]
@mathstuf Can you rebase?  That should pull in some changes which fix the broken unit tests.

**Comment by mathstuf** [comment]
Rebased.

**Comment by carllerche** [comment]
Thanks for pinging me... Sorry for letting this slip.

**Pull request merged**